Marlin/src/module/motion.cpp

```diff
@@ -94,7 +94,11 @@ bool relative_mode; // = false;
  *   Used by 'line_to_current_position' to do a move after changing it.
  *   Used by 'sync_plan_position' to update 'planner.position'.
  */
-xyze_pos_t current_position = { X_HOME_POS, Y_HOME_POS, Z_HOME_POS };
+#ifdef Z_IDLE_POS
+  xyze_pos_t current_position = { X_HOME_POS, Y_HOME_POS, Z_IDLE_POS };
+#else
+  xyze_pos_t current_position = { X_HOME_POS, Y_HOME_POS, Z_HOME_POS };
+#endif
```

Comment:
Could this work?
`xyze_pos_t current_position = { X_HOME_POS, Y_HOME_POS, TERN(IF_ENABLED(Z_IDLE_POS), Z_IDLE_POS , Z_HOME_POS) };`

Reply:
I believe TERN only works with constants that act as boolean options. They can be undefined, 0, or 1.

Reply:
Yea, my bad, `IF_ENABLED` would give null if disabled.
Is there a macro that gives 1 if enabled and 0 if disabled? I still think that code should be shrunken down with macro.

Reply:
Since the introduction of the TERN macros you will see patterns like this in various places:

```
#ifdef Z_IDLE_POS
   #define HAS_Z_IDLE_POS 1
#endif
...
TERN(HAS_Z_IDLE_POS, Z_IDLE_POS, Z_HOME_POS);
```

That first portion might be in the file that needs it, or it might be off somewhere like `Conditionals_post.h`.

Reply:
I tried to solve it with TERN first and failed 😄 .

> ```
> #ifdef Z_IDLE_POS
>    #define HAS_Z_IDLE_POS 1
> #endif
> ...
> TERN(HAS_Z_IDLE_POS, Z_IDLE_POS, Z_HOME_POS);
> ```

@sjasonsmith Thanks for that. I will implement it later on.